sqlmesh/cli/main.py

```diff
@@ -426,20 +406,24 @@ def create_external_models(obj: Context) -> None:
 @click.option(
     "--limit",
     type=int,
+    default=20,
     help="The limit of the sample dataframe.",
 )
 @click.pass_obj
 @error_handler
-def table_diff(obj: Context, **kwargs: t.Any) -> None:
-    """Show the diff between two tables.
-
-    Can either be two tables or two environments and a model.
-    """
-    kwargs["model_or_snapshot"] = kwargs.pop("model", None)
-    on = kwargs.pop("on", None)
-    grain = ensure_list(kwargs.pop("grain", None))
-    kwargs["on"] = exp.condition(on) if on else grain
-    obj.table_diff(**kwargs)
+def table_diff(
+    obj: Context, from_to: str, model_name: str, on: t.Optional[str], **kwargs: t.Any
+) -> None:
+    """Show the diff between two tables."""
+    model = obj.get_model(model_name, raise_if_missing=True)
+    source, target = from_to.split(":")
+    obj.table_diff(
+        source=source,
+        target=target,
+        model_or_snapshot=model,
+        on=exp.condition(on or " and ".join([f"s.{col} = t.{col}" for col in model.grain])),
```

Comment:
Lets evaluate assert either `on` or `model.grain` is truthy before this line so we can provide a nicer exception message to the user to easily resolve.

Reply:
you shouldn't have to do this for model.grain, it's already built into base table diff